tests/pipelines/test_pipeline.py

```diff
@@ -266,27 +266,22 @@ def test_pipeline_get_artifacts():
     pipeline.run("train")
     pipeline.run("register")
 
-    assert isinstance(pipeline.get_artifact("ingested_data"), pd.DataFrame)
-    assert isinstance(pipeline.get_artifact("training_data"), pd.DataFrame)
-    assert isinstance(pipeline.get_artifact("validation_data"), pd.DataFrame)
-    assert isinstance(pipeline.get_artifact("test_data"), pd.DataFrame)
-    assert isinstance(pipeline.get_artifact("transformed_training_data"), pd.DataFrame)
-    assert isinstance(pipeline.get_artifact("transformed_validation_data"), pd.DataFrame)
-    assert hasattr(pipeline.get_artifact("transformer"), "transform")
-    assert isinstance(pipeline.get_artifact("model"), mlflow.pyfunc.PyFuncModel)
-    assert isinstance(pipeline.get_artifact("run"), Run)
-    assert isinstance(pipeline.get_artifact("registered_model_version"), ModelVersion)
+    assert isinstance(pipeline.get_artifact("ingested_data").load(), pd.DataFrame)
```

Comment:
Since we are changing the behavior of `get_artifact` I think we should document this. Can we add a release note, and maybe change the wording of the `get_artifact` docstring?

Reply:
Is there a reason we are making this backwards compact change?
We might need to update the notebook examples in template and example repo for regression template.

Reply:
Good point, yes the notebooks would need to be updated. My understanding is that the cost of breaking changes is small given lack of MLP usage, so the fact that it's a breaking change didn't really affect the direction of this PR. Do you disagree with that sentiment and/or do you prefer the previous interface?

Reply:
I would surely prefer the previous interface since we don't need to call load additional from users perspective. 

Users just need to call `pipeline.get_artifact("ingested_data")` for example. Why is load important? 

Unless we can plan to add more function like
`pipeline.get_artifact("ingested_data").clean()`, `pipeline.get_artifact("ingested_data").update()`, etc.
For me adding the load doesn't make a lot of sense for the added complexity for the user. Just my thoughts.


What do you guys think?

Reply:
Thank you for making this change. Love the refactor part :)